docs/checkit.js

```diff
@@ -438,22 +441,175 @@ CheckIt.prototype.removeEmptyWeeks = function(calObj, $calendarDiv) {
     })
 };
 
-CheckIt.prototype.collapseBuildMenu = function() {
-    // Collapse the build calendar form.
-    this.$buildCalendarForm.collapse('hide');
+CheckIt.prototype.hideForm = function($form) {
+    // Collapse a collapsible div (in checkit's case forms).
+    // Parameter $form is the selector for a collapsible form (or div).
+    
+    $form.collapse('hide');
 };
 
-CheckIt.prototype.uncollapseBuildMenu = function() {
-    // Show the build calendar form.
-    this.$buildCalendarForm.collapse('show');
+CheckIt.prototype.showForm = function($form) {
+    // Show a collapsible form.
+    // Parameter $form is the selector for a collapsible form (or div).
+    
+    $form.collapse('show');
 };
 
-CheckIt.prototype.clearForm = function() {
+CheckIt.prototype.clearForm = function($fullForm) {
     // Clears the buildCalendar form.
+    // $fullForm is the selector for the form div
+    // Only removeFormErrors if there are any errors on display
+    if ($fullForm.find('.has-error')) {
+        this.removeFormErrors($fullForm);
+    }
+    $fullForm.trigger("reset"); 
+};
+
+CheckIt.prototype.addFieldError = function($id, $srId) {
+
+    // Set the appropriate CSS on the form field to indicate an error state
+    // $id is the selector for the div with class .form-group that
+    // surrounds the input field with the error
+    // $srId is the selector for the screen reader element
+    $id.addClass('has-error has-feedback');
+    $srId.removeClass('hidden');
     
-    // TODO only removeFormErrors if there are any errors on display
-    this.removeFormErrors();
-    this.$fullForm[0].reset();
+};
+
+CheckIt.prototype.removeFieldError = function($id, $srId) {
+    // Removes error state CSS on form field 
+    // $id is the selector for the div with class .form-group that
+    // surrounds the input field with the error
+    // $srId is the selector for the screen reader element
+    $id.removeClass('has-error has-feedback');
+    $srId.addClass('hidden');
+};
+
+CheckIt.prototype.addHelpBlock = function($helpBlock) {
```

Comment:
Looks like these four add/remove functions can be combined to just two show and hide functions.